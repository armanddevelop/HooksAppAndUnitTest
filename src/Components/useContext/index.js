export { HomeScreen } from "./HomeScreen";
export { AboutScreen } from "./AboutScreen";
export { LognScreen } from "./LognScreen";
export { NavBar } from "./NavBar";
