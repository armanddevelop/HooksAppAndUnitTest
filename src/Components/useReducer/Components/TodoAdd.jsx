import React from "react";
import { useForm } from "../../../Hooks/useForm";
import PropTypes from "prop-types";
export const TodoAdd = ({ handleAddToDo }) => {
  const { formState, handleInputChange, resetField } = useForm({
    description: "",
  });
  const { description } = formState;
  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length === 0) return;
    const newTodo = {
      id: new Date().getTime(),
      description,
      done: false,
    };
    handleAddToDo(newTodo);
    resetField();
  };
  return (
    <>
      <h4>Add Task</h4>
      <hr />
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="description"
          placeholder="learn code"
          autoComplete="off"
          className="form-control"
          value={description}
          onChange={handleInputChange}
        ></input>
        <button
          type="submit"
          className="btn btn-outline-primary mt-1 btn-block"
        >
          ADD
        </button>
      </form>
    </>
  );
};
TodoAdd.propTypes = {
  handleAddToDo: PropTypes.func.isRequired,
};
