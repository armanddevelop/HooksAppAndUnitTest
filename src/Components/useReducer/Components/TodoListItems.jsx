import PropTypes from "prop-types";
export const TodoListItems = ({
  description,
  done,
  idx,
  id,
  handleDelete,
  handleToogle,
}) => {
  return (
    <li className="list-group-item">
      <p
        className={`${done ? "complete" : ""}`}
        onClick={() => handleToogle(id)}
      >
        {idx + 1}.- {description}
      </p>
      <button onClick={() => handleDelete(id)} className="btn btn-danger">
        Delete
      </button>
    </li>
  );
};
TodoListItems.prototype = {
  description: PropTypes.string.isRequired,
  done: PropTypes.bool.isRequired,
  id: PropTypes.number.isRequired,
  idx: PropTypes.number.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToogle: PropTypes.func.isRequired,
};
