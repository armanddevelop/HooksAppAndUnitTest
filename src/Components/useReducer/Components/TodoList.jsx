import { TodoListItems } from "./TodoListItems";
import PropTypes from "prop-types";

export const TodoList = ({ todos, handleDelete, handleToggle }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map(({ id, description, done }, idx) => {
        return (
          <TodoListItems
            key={id}
            description={description}
            done={done}
            idx={idx}
            id={id}
            handleDelete={handleDelete}
            handleToogle={handleToggle}
          />
        );
      })}
    </ul>
  );
};
TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle: PropTypes.func.isRequired,
};
