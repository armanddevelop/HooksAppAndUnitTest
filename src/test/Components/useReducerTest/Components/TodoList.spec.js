import { TodoList } from "../../../../Components/useReducer/Components/TodoList";
import { demosTodos } from "../../../fixtures/demosTodos";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";

describe("Unit Test to TodoList component", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const container = shallow(
    <TodoList
      todos={demosTodos}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  it("Snapshot to TodoList", () => {
    expect(toJson(container)).toMatchSnapshot();
  });

  it("TodoListItem component exist", () => {
    expect(container.find("TodoListItems").exists()).toBeTruthy();
  });

  it("Should have TodoListItem", () => {
    expect(container.find("TodoListItems").length).toBe(demosTodos.length);
    //console.log(container.find("TodoListItems").at(0).props());
    const { description, done, id, handleDelete, handleToogle } = container
      .find("TodoListItems")
      .at(0)
      .props();
    expect(description).toBe(demosTodos[0].description);
    expect(done).toBe(demosTodos[0].done);
    expect(id).toBe(demosTodos[0].id);
    expect(handleDelete).toEqual(expect.any(Function));
    expect(handleToogle).toEqual(expect.any(Function));
  });
});
