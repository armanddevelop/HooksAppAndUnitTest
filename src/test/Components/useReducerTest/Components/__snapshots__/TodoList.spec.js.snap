// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test to TodoList component Snapshot to TodoList 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItems
    description="aprender a hacer unit test in React"
    done={false}
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    id={1}
    idx={0}
    key="1"
  />
  <TodoListItems
    description="aprender mongo"
    done={false}
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    id={2}
    idx={1}
    key="2"
  />
</ul>
`;
